### PR TITLE
Dialog warnings for using MakeCode with a V1

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

**Comment by microbit-robert** [comment]
Some changes to messages/ui.en.json are the result of running the i18n compile step and not directly related to this PR.

File: src/components/IncompatibleEditorDevice.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                      />
+                    </Text>
+                  </>
+                ) : (
+                  <Text>
+                    <FormattedMessage
+                      id="incompatible-device-body-alt"
+                      values={{
+                        link: (chunks: ReactNode) => (
+                          <Text
```
Comment (microbit-matt-hillsdon) [inline src/components/IncompatibleEditorDevice.tsx:83]: I think these are Button variant=link elsewhere. Would that work here too?

File: src/components/TestingModelGridView.tsx [hunk 2]
```diff
@@ -70,8 +74,32 @@ const TestingModelGridView = () => {
...
+  }, [onClose, openEditor]);
+
+  const maybeOpenEditor = useCallback(() => {
+    // Open editor if device is not a V1, otherwise show warning dialog.
+    const bluetoothConnection = getBluetoothConnection();
+    if (bluetoothConnection) {
+      if (bluetoothConnection.getBoardVersion() === "V1") {
+        return onOpen();
+      }
+    }
```
Comment (microbit-matt-hillsdon) [inline src/components/TestingModelGridView.tsx:91]: Might be nice to hide the bluetooth-specific nature of this behind something in the hook as we might do V1 radio bridge at some point.

File: src/messages/ui.en.json [hunk 3]
```diff
@@ -1970,7 +2004,7 @@
   "unplug-radio-link-microbit-description": [
     {
       "type": 0,
-      "value": "Unplug the radio link micro:bit from the computer. If you want to record more data samples, you will need to reconnect with the tool."
+      "value": "Unplug the radio link micro:bit from the computer. You will need to reconnect the micro:bit if you want to record more data samples."
```
Comment (microbit-matt-hillsdon) [inline src/messages/ui.en.json:2007]: Ah yeah, this was a last minute copy tweak on another PR.

Comment (microbit-matt-hillsdon) [reply src/messages/ui.en.json:2007]: Similarly the removals.

File: src/components/TestingModelGridView.tsx (hunk 2 above)
Comment (microbit-robert) [reply src/components/TestingModelGridView.tsx]: Done here. A bit more work to hold on to the radio remote board version [5dc85ea](https://github.com/microbit-foundation/ml-trainer/pull/358/commits/5dc85ea44d2b03748b9b729314d8a38b64e233c4)

`radioRemoteBoardVersion` is currently not cleared if you go back to the home page and start a new session. It's only cleared on a page refresh.

File: src/components/IncompatibleEditorDevice.tsx (hunk 1 above)
Comment (microbit-robert) [reply src/components/IncompatibleEditorDevice.tsx]: Done here. Updated another instance as well [a0adffe](https://github.com/microbit-foundation/ml-trainer/pull/358/commits/a0adffe6268b08a55eaeee30f6a562b51dfb8972)

File: src/components/TestingModelGridView.tsx (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply src/components/TestingModelGridView.tsx]: We're going to ponder this on a separate ticket.

**Comment by microbit-matt-hillsdon** [comment]
Just waiting on copy feedback now, otherwise LGTM.

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Comment by microbit-matt-hillsdon** [comment]
Copy feedback not likely in time for beta, gone with best guess and will tweak later.

**Pull request merged**